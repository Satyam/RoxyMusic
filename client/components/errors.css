.errors-list {
  composes: global(alert);
  composes: global(alert-warning);
  composes: global(alert-dismissible);
  white-space: pre;
  font-family: monospace;
}

.closeButton {
  composes: global(close);
  composes: global(pull-right);
}
.closeButton::before {
  content: "\00D7";
}

:global .btn {
  margin-left: 0.5em;
  margin-right: 0.5em;
}
