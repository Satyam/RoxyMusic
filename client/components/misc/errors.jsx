import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import isPlainClick from '_utils/isPlainClick';
import classNames from 'classnames';
import { clearHttpErrors } from '_store/actions';
import styles from './errors.css';

export const ErrorsComponent = ({ errors, onCloseErrors }) => {
  const closeErrorsHandler = ev => isPlainClick(ev) && onCloseErrors();
  return (
    <div
      className={classNames(
        'errors',
        styles.errorsList,
        { hide: !errors.length }
      )}
    >
      <button onClick={closeErrorsHandler} className={styles.closeButton} />
      {
        errors.map(
          process.env.NODE_ENV === 'production'
          ? e => e.message
          : e => JSON.stringify(e, null, 2)
        ).join('\n')
      }
    </div>
  );
};

ErrorsComponent.propTypes = {
  errors: PropTypes.array,
  onCloseErrors: PropTypes.func,
};

export const mapStateToProps = state => ({
  errors: state.requests.errors,
});

export const mapDispatchToProps = dispatch => ({
  onCloseErrors: () => dispatch(clearHttpErrors()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ErrorsComponent);
