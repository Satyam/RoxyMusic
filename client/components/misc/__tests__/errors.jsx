import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import { ErrorsComponent } from '../errors';

describe('ErrorsComponent', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ErrorsComponent errors={[]} />, div);
  });
  it('enzyme shallow', () => {
    const err = shallow(<ErrorsComponent errors={[]} />);
    console.log(err.text());
  });
});
