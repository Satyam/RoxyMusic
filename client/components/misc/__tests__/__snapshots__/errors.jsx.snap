// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorsComponent snapshot testing with errors property 1`] = `
<div
  className="errors"
>
  <button
    onClick={[Function]}
  />
  {
    "error": {
      "message": "whatever"
    }
  }
  {
    "error": {
      "message": "another"
    }
  }
</div>
`;

exports[`ErrorsComponent snapshot testing with no properties 1`] = `
<div
  className="errors hide"
>
  <button
    onClick={[Function]}
  />
</div>
`;
